File: SingletonDesignPattern/Counter_Example/Driver.java
package SingletonDesignPattern.Counter_Example;

public class Driver {
  public static void main(String[] args) {

    Thread thread1 = new Thread(() -> {
      Counter counter1 = Counter.getInstance();
      for (int i = 0; i < 5; i++) {
        counter1.increament();
      }
      System.out.printf("Count value using counter1: %d\n", counter1.getCount());
    });

    Thread thread2 = new Thread(() -> {
      Counter counter2 = Counter.getInstance();
      for (int i = 0; i < 10; i++) {
        counter2.increament();
      }
      System.out.printf("Count value using counter2: %d\n", counter2.getCount());
    });

    thread1.start();
    thread2.start();
  }
}

File: SingletonDesignPattern/Counter_Example_Python/SOL1_DoubleCheckLocking/Driver.py
from Counter import Counter

class Driver():

  @staticmethod
  def main() -> None:

    counter1: Counter = Counter.getInstance()
    counter2: Counter = Counter.getInstance()

    print(f"Is the counter1 == counter2 ?: {counter1 == counter2}")
    print(f"The count from both the Counter Objects counter1: {counter1.getCount()} and counter2: {counter2.getCount()}")

    for _ in range(5):
      counter1.increment()
      print(f"Increamenting the count using the counter1 obj but getting the count using counter2 obj: {counter2.getCount()}")

    print(f"Count using counter1 obj: {counter1.getCount()}")


if __name__ == "__main__":

  Driver.main()